docs/getting_started/typescript/hello_world_in_typescript/index.md

```diff
@@ -234,316 +317,405 @@ async function run() {
 
-  before(async () => {
-    testEnv = await TestWorkflowEnvironment.createLocal();
-  });
+The project generator created a `client.ts` file in the `src` directory, but it will need a couple changes to make it work for this application:
 
-  after(async () => {
-    await testEnv?.teardown();
-  });
+- You'll need to change the name of the Workflow.
+- You'll need to add code that lets you specify the person's name as the command-line argument so it can be passed to the Workflow, instead of using the hard-coded value of `Temporal`.
 
-  it('successfully completes the Workflow', async () => {
-    const { client, nativeConnection } = testEnv;
-    const taskQueue = 'test';
+Open the `src/client.ts` file in your editor and modify it so it looks like the following:
 
-    const worker = await Worker.create({
-      connection: nativeConnection,
-      taskQueue,
-      workflowsPath: require.resolve('../workflows'),
-      activities,
-    });
+```typescript
+import { Connection, Client } from '@temporalio/client';
+import { getAddressFromIP } from './workflows';
+import { nanoid } from 'nanoid';
+import { TASK_QUEUE_NAME } from './shared';
+import process from 'process';
 
-    const result = await worker.runUntil(
-      client.workflow.execute(example, {
-        args: ['Temporal'],
-        workflowId: 'test',
-        taskQueue,
-      })
-    );
-    assert.equal(result, 'Hello, Temporal!');
+async function run() {
+
+  if (process.argv.length <= 2) {
+    console.error('Must specify a name as the command-line argument');
+    process.exit(1);
+  }
+
+  const name = process.argv[2];
+  const connection = await Connection.connect({ address: 'localhost:7233' });
+  const client = new Client({ connection });
+
+  const handle = await client.workflow.start(getAddressFromIP, {
+    taskQueue: TASK_QUEUE_NAME,
+    args: [name],
+    workflowId: 'getAddressFromIP-' + nanoid(),
   });
+  console.log(await handle.result());
+}
+
+run().catch((err) => {
+  console.error(err);
+  process.exit(1);
 });
+
 ```
-<!--SNIPEND-->
 
-This test creates a test execution environment and then registers the Activity to the Worker. The test then executes the Workflow in the test environment and checks for a successful execution. Finally, the test ensures the Workflow's return value returns the expected value.
+You import the `process` module so you can access command-line arguments you'll pass.
 
-Run the following command from the project root to execute the unit tests:
+In the `run` function you check to see if there is at least two arguments passed. You need to check for two arguments because `process.argv` includes the Node.js executable path and the script path as the first two elements, so the actual command-line arguments start at index 2.
```

Comment:
```suggestion
In the `run` function you check to see if there are at least two arguments passed. You need to check for two arguments because `process.argv` includes the Node.js executable path and the script path as the first two elements, so the actual command-line arguments start at index 2.
```